cEP-0028.md

```diff
@@ -0,0 +1,150 @@
+   extension for bears would be necessary. These parameters enable support
+   for hooking a section for a particular tag type. For example:
+
+   ```
+    [longrunning]
+    enabled = True
+    tags = cls, save
+    bears = SomeLongRunningBear
+   ```
+
+   This kind of a configuration will enable the section completely during
+   analysis cycles of coala via a language server while also enabling it for
+   regular invocations.
+
+   ```
+    [longrunning]
+    tags = cls
+    files = *.py
+    bears = SomeLongRunningBear
+
+    [longrunning.some]
+    tags += save, change
+    bears = SomeLongRunningBear
+   ```
+
+   This example shows a case where the sections inherit the tags and
+   thereby `longrunning.some` runs on both `save` and `change` tagged
+   execution cycles.
+
+   To enable tagged executional support, coala-core needs to be extended
+   and at least support a new optional INPUT cli argument taking list of
+   tags to be executed.
+
+   ### Supported Tags
+
+   The following are the supported default tags groups:
+
+   - `cls`: A tag indicating if the section should be enabled for coala
+     language server. This should be used in conjunction with other
+     available tags.
+
+   - `open`: Tag group to execute during a open event on a file.
+
+   - `change`: Tag group to execute during a change event on the
+      project files.
+
+   - `save`: Tag group to execute on a save event on the project files.
+
+   - `formatting`: Tag group to execute when a `formatting` event is
+      raised on a file.
+
+2. Custom entry point into coala core
+   Since coala core was not designed to be a long running process coalib
+   does not provide a flexible entry point into it. To maximize the efficiency
+   of the coala language server, it will require an entry point that supports:
+
+   - Support for `FileProxies`: Since not all changes on the editor are flushed
+     to the disk coala core will have to support processing file content
+     directly as opposed to files.
+
+   - Support for preprocessing coala configurations: The configurations for
+     coala analysis need to be generated at runtime for each analysis cycle
+     based upon the `global type bear` status of the bears.
+
+   The following functions are core to the current general cli entry point and
+   need to be patched as part of the custom coala language server entry point
+   for:
+
+   - `coalib.processes.Processing.create_file_dict()`
+   - `coalib.processes.Processing.execute_section()`
+   - `coalib.coala_main.run_coala()`
+
+   The prototypes for functions that will replace/patch those are:
+
+   ```python
+    def create_file_dict(filename_list,
+                         fileproxy_map,
+                         log_printer=None):
+        """
+        Reads all files into a dictionary.
+        :param filename_list:   List of names of paths to files to get contents of.
+        :param fileproxy_map:   FileProxyMap containing the files available.
```

Comment:
We need to see what this `FileProxyMap` class is like

Reply:
PTAL https://github.com/ksdme/coala-ls/blob/5d4c230cd1a819572d423a5ba90a537d0ca1b91c/coalals/utils/files.py#L209